src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/service/CertifiedCopyItemService.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.companieshouse.certifiedcopies.orders.api.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.certifiedcopies.orders.api.model.CertifiedCopyItem;
+import uk.gov.companieshouse.certifiedcopies.orders.api.model.DeliveryMethod;
+import uk.gov.companieshouse.certifiedcopies.orders.api.repository.CertifiedCopyItemRepository;
+
+import java.time.LocalDateTime;
+
+@Service
+public class CertifiedCopyItemService {
+
+    private final CertifiedCopyItemRepository repository;
+    private final EtagGeneratorService etagGenerator;
+    private final LinksGeneratorService linksGenerator;
+    private final IdGeneratorService idGenerator;
+
+    public CertifiedCopyItemService(final CertifiedCopyItemRepository repository,
+                                    final EtagGeneratorService etagGenerator,
+                                    final LinksGeneratorService linksGenerator,
+                                    final IdGeneratorService idGenerator) {
+        this.repository = repository;
+        this.etagGenerator = etagGenerator;
+        this.linksGenerator = linksGenerator;
+        this.idGenerator = idGenerator;
+    }
+
+    public CertifiedCopyItem createCertifiedCopyItem(final CertifiedCopyItem certifiedCopyItem) {
+        final LocalDateTime now = LocalDateTime.now();
+        certifiedCopyItem.setCreatedAt(now);
+        certifiedCopyItem.setUpdatedAt(now);
+        certifiedCopyItem.setId(idGenerator.autoGenerateId());
+        certifiedCopyItem.setEtag(etagGenerator.generateEtag());
+        certifiedCopyItem.setLinks(linksGenerator.generateLinks(certifiedCopyItem.getId()));
+
+        if(certifiedCopyItem.getData().getItemOptions().getDeliveryMethod().equals(DeliveryMethod.POSTAL)) {
+            certifiedCopyItem.getData().setPostalDelivery(Boolean.TRUE);
+        } else {
+            certifiedCopyItem.getData().setPostalDelivery(Boolean.FALSE);
+        }
+        certifiedCopyItem.getData().setKind("item#certified-copy");
```

Comment:
I note we have dispensed with the `ItemType` `enum` here. Not really necessary?

Reply:
I copied over what we had from the certificates api. The kind in that project is a string (there is no ItemType enum)

Reply:
Well it may have been unnecessary overhead, but there is an `ItemType`?: https://github.com/companieshouse/certificates.orders.api.ch.gov.uk/blob/af637b304a1256fcd53f12401909cdbfac717b62/src/main/java/uk/gov/companieshouse/certificates/orders/api/ItemType.java. Perhaps that can be examined as part of whichever story will deal with the description.

Reply:
Oh I completely missed that. I was expecting it to be in the model folder! I agree with you, we can have a look at it in GCI-1207.